aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/CreateHandler.java

```diff
@@ -74,24 +77,19 @@ private void setPrimaryIdentifier(ResourceHandlerRequest<ResourceModel> request,
                 logger.log(String.format("%s successfully created.", ResourceModel.TYPE_NAME));
                 return awsResponse;
             })
-            .stabilize((awsRequest, awsResponse, client, model, context) -> isStabilized(model, client))
-            .progress();
+            .done((awsRequest, response, client, model, context) -> {
+                try {
+                    logger.log(String.format("waiting %s milliseconds for repository to become available", PROPAGATION_DELAY_MS));
+                    Thread.sleep(PROPAGATION_DELAY_MS);
+                    return ProgressEvent.progress(model, context);
+                } catch (InterruptedException e) {
+                    throw new CfnGeneralServiceException(e);
+                }
+            });
     }
 
     private boolean hasReadOnlyProperties(final ResourceModel model) {
         return model.getName() != null;
     }
 
-    private boolean isStabilized(
-        final ResourceModel model,
-        final ProxyClient<CodeartifactClient> proxyClient
-    ) {
-        try {
-            proxyClient.injectCredentialsAndInvokeV2(
-                    Translator.translateToReadRequest(model), proxyClient.client()::describeRepository);
-            return true;
-        } catch (ResourceNotFoundException e) {
-            return false;
-        }
-    }
```

Comment:
we still want to call this, no? Like, we want to know after that 10s of sleeping, our resource is actually there

Reply:
hmm yeah thats true, I assumed that it will definitely be there after sleep, but you are right. I'll update